toolchain/check/cpp/macros.cpp

```diff
@@ -60,13 +61,21 @@ auto TryEvaluateMacroToConstant(Context& context, SemIR::LocId loc_id,
   }
 
   clang::Expr::EvalResult evaluated_result;
-  if (!result_expr->EvaluateAsInt(evaluated_result, sema.getASTContext())) {
-    context.TODO(loc_id, "non-integer constant expression in macro.");
-    return nullptr;
+  if (result_expr->EvaluateAsInt(evaluated_result, sema.getASTContext())) {
+    return clang::IntegerLiteral::Create(
+        sema.getASTContext(), evaluated_result.Val.getInt(),
+        result_expr->getType(), result_expr->getExprLoc());
+  }
+  llvm::APFloat ap_value(0.0);
```

Comment:
Just to note, if this remains (might be moot for below comment), there's APInt too so probably better to say "float" instead of just "value"
```suggestion
  llvm::APFloat ap_float(0.0);
```

Reply:
This became obsolete. PTAL.